relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -1,20 +1,473 @@
+    fn from(value: TransactionsMetric) -> Self {
+        let (ty, unit, name, value, timestamp, tags) = match value {
+            TransactionsMetric::User {
+                value,
+                timestamp,
+                tags,
+            } => (
+                MetricType::Set,
+                MetricUnit::None,
+                "user".to_string(),
+                MetricValue::set_from_str(&value),
+                timestamp,
+                tags.into(),
+            ),
+            TransactionsMetric::Breakdowns {
+                timestamp,
+                value,
+                tags,
+                name,
+            } => (
+                MetricType::Distribution,
+                MetricUnit::Duration(DurationUnit::MilliSecond),
+                format!("breakdowns.{name}"),
+                MetricValue::Distribution(value),
+                timestamp,
+                tags.into(),
+            ),
+            TransactionsMetric::CountPerRootProject {
+                value,
+                timestamp,
+                tags,
+            } => (
+                MetricType::Counter,
+                MetricUnit::None,
+                "count_per_root_project".to_string(),
+                MetricValue::Counter(value),
+                timestamp,
+                tags.into(),
+            ),
+            TransactionsMetric::Duration {
+                unit,
+                value,
+                timestamp,
+                tags,
+            } => (
+                MetricType::Distribution,
+                MetricUnit::Duration(unit),
+                "duration".to_string(),
+                MetricValue::Distribution(value),
+                timestamp,
+                tags.into(),
+            ),
+            TransactionsMetric::Measurements {
+                kind,
+                value,
+                timestamp,
+                unit,
+                tags,
+            } => (
+                MetricType::Distribution,
+                unit,
+                format!("measurements.{kind}"),
+                MetricValue::Distribution(value),
+                timestamp,
+                tags.into(),
+            ),
+        };
+
+        let mri = MetricResourceIdentifier {
+            ty,
+            namespace: MetricNamespace::Transactions,
+            name,
+            unit,
+        };
+
+        Metric::new(mri, value, timestamp, tags)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+struct TransactionsMeasurementTags {
+    measurement_rating: Option<String>,
+    release: Option<String>,
+    dist: Option<String>,
+    environment: Option<String>,
+    transaction: Option<String>,
+    platform: Option<String>,
+    transaction_status: Option<String>,
+    transaction_op: Option<String>,
+    http_method: Option<String>,
+    other: BTreeMap<String, String>,
+}
+
+impl TransactionsMeasurementTags {
+    fn new(measurement_rating: Option<String>, universal_tags: UniversalTags) -> Self {
```

Comment:
If we're keeping `UniversalTags` (which I think makes sense because of `extract_universal_tags`, let's store it in a nested fashion too, e.g.

```rust
struct TransactionMeasurementTags {
    universal_tags: UniversalTags,
    measurement_rating: Option<String>,
}
```